### PR TITLE
Code design refactor

### LINKED ISSUE #29: Refactoring Story One.
Related: #18

### PR DESCRIPTION
**Pull request by RogerDMK**
# Description
Refactored code to improve the overall design, specifically the implementation of routes.

Fixes #29 
mainRoutes has been modified, moving routes to gameRoutes, userRoutes and lobbyRoutes, all relevant changes to other files have been made

### CODE REVIEW

File: src/services/wordQuery.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+              } else {
+                resolve(JSON.stringify({ message: `${word} has been saved to the database, you NONCE that's the wrong word! D:<` }))
+              }
+            }).catch(reject)
+          } else {
+            resolve(JSON.stringify({ message: `There was an error saving ${word} to the database` }))
+          }
+        })
+      } catch (err) {
+        console.log(err)
```
Comment (UntidyRAM) [inline src/services/wordQuery.js:59]: Should this not be a reject?

File: src/routes/lobbyRoutes.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+const { createGame, prevGameID } = require('../services/lobby.js')
+
+lobbyRouter.post('/', async function (req, res) { // ?
+    res.sendFile(path.join(__dirname, '../views', 'lobby.html'))
+})
+
+lobbyRouter.post('/create', async function (req, res) { // ?
+    const prevResult = await prevGameID()
+    const newID = parseInt(prevResult) + 1
+    const result = await createGame(req.body.gameModeInput, newID)
```
Comment (UntidyRAM) [inline src/routes/lobbyRoutes.js:17]: Doesn't the database gameID field auto-increment? So surely we wouldn't need to supply the value for the insertion statement? I know this isn't part of this review, I'm just putting it here for future reference 😆

File: src/services/wordQuery.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+async function createWord (word) {
+  return new Promise((resolve, reject) => {
+    const sqlCode = `INSERT INTO [HazardaGuess_db].[dbo].[WordLog] (Word) VALUES ('${word}');`
+    get('default').then(
+      (pool) => pool.request().query(sqlCode).then(
+        (result) => {
+          resolve(result)
+        }
+      )// .catch(reject(console.error)) // TODO: find out why this is causing the server to crash
```
Comment (UntidyRAM) [inline src/services/wordQuery.js:13]: Need to investigate why the catch is causing an error. Maybe replacing it with ```.catch(reject)``` might fix it?

File: src/services/wordQuery.js [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+module.exports = {
+  createWord: createWord,
+  checkWord: checkWord,
+  makeGuess: async function makeGuess(word, gameId){
+    return new Promise((resolve, reject) => {
+    if (/^[a-zA-Z]+$/.test(word) === true & word.length === 5) {
+      try {
+        console.log('hello')
```
Comment (UntidyRAM) [inline src/services/wordQuery.js:39]: Hi?

**Review by UntidyRAM — APPROVED** [review]
Looks good. I just highlighted a few very small issues that can be fixed later. Nice job ;)

File: src/routes/lobbyRoutes.js (hunk 2 above)
Comment (RogerDMK) [reply src/routes/lobbyRoutes.js:17]: This is how jess set up the code, I agree that it should auto increment, but the database should be changed before we add that change

File: src/services/wordQuery.js (hunk 4 above)
Comment (RogerDMK) [reply src/services/wordQuery.js:39]: Whoops debugging left in.... Will remove it

File: src/services/wordQuery.js (hunk 1 above)
Comment (RogerDMK) [reply src/services/wordQuery.js:59]: Yes, forgot to change it when converting to a promise.

File: src/services/wordQuery.js (hunk 3 above)
Comment (RogerDMK) [reply src/services/wordQuery.js:13]: Probably, I think that is someone else's error

Comment (RogerDMK) [reply src/services/wordQuery.js:13]: Oh in the reject you should not write to the console, just a reject

**Pull request merged**